### PR TITLE
Soen 109 add/delete user when changing role and approving

### PR DESCRIPTION
**Pull request by jfv-2000**
# Description

Implements/fixes [SOEN-109](https://soen390team8.atlassian.net/jira/software/projects/SOEN/boards/1?selectedIssue=SOEN-109)

- Delete existing patient/doctor data in table if role is changed
- Add new patient/doctor data in table if it is the new role given
- Add doctor data in table after being approved by the admin
- Front-end fix so that only the button can be used to approve a user and not the whole user card

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Unit testing

### CODE REVIEW

**Review by stevenvuong00 — APPROVED** [review]
great work as always! good bug fixes

**Review by silentswordxv — APPROVED** [review]
Thanks for catching this!

**Review by KelvinTrang — APPROVED** [review]
good work!

File: backend/__tests__/routes/admins.test.js [hunk 1]
```diff
@@ -85,7 +84,7 @@ describe("PATCH: Update user role", () => {
             },
             {
                 where: {
-                    AccountId: "51",
+                    AccountId: "828",
```
Comment (NicholasYip) [inline backend/__tests__/routes/admins.test.js]: should we move this id to a constant? I see that you use it again lower

File: backend/controllers/admin-controller.js [hunk 2]
```diff
@@ -146,14 +177,24 @@ function confirmAccount(req, res) {
...
         }
-    )
-        .then(() => {
-            res.status(200).send("Account successfully confirmed !");
-        })
-        .catch(err => {
-            console.log("[Approve-User] Error: ", err);
-            res.status(400).send("Failed to confirm account");
+    ).catch(err => {
+        console.log("[Approve-User] Error: ", err);
```
Comment (NicholasYip) [inline backend/controllers/admin-controller.js:181]: should this log be removed?

**Review by NicholasYip — APPROVED** [review]
nice work

File: backend/__tests__/routes/admins.test.js (hunk 1 above)
Comment (jfv-2000) [reply backend/__tests__/routes/admins.test.js]: great idea !

**Pull request merged**